### PR TITLE
Refactor cleanup, firehol, and mass_scanners cronjobs to use repositories. Addresses #633

### PR DESCRIPTION
**Pull request by opbot-xd**
# Description

This PR implements Phase 2 of the repository pattern refactoring by migrating the `cleanup.py`, `firehol.py`, and `mass_scanners.py` cronjobs to use the repository pattern instead of direct Django ORM calls.

### New Repositories Created:
1. **FireHolRepository** (`greedybear/cronjobs/repositories/firehol.py`)
   - `get_or_create()`: Get existing or create new FireHol blocklist entry
   - `save()`: Persist FireHol entry
   - `delete_old_entries()`: Delete entries older than specified date
   - `cleanup_old_entries()`: Cleanup with configurable retention days (default 30)

2. **MassScannerRepository** (`greedybear/cronjobs/repositories/mass_scanner.py`)
   - `get_by_ip()`: Retrieve mass scanner by IP address
   - `create()`: Create new mass scanner entry
   - `save()`: Persist mass scanner entry
   - `exists()`: Check if mass scanner exists by IP

### Extended Existing Repositories:

1. **IocRepository** - Added cleanup methods:
   - `delete_old_iocs()`: Delete IOC records older than cutoff date
   - `update_ioc_reputation()`: Update IP reputation for existing IOC

2. **CowrieSessionRepository** - Added cleanup methods:
   - `delete_old_command_sequences()`: Delete old command sequences
   - `delete_incomplete_sessions()`: Delete sessions without start_time
   - `delete_sessions_without_login()`: Delete old sessions without login attempts
   - `delete_sessions_without_commands()`: Delete old sessions without commands

### Refactored Cronjobs:

1. **CleanUp** (`greedybear/cronjobs/cleanup.py`)
   - Now uses `IocRepository` and `CowrieSessionRepository`
   - Replaced all direct `IOC.objects`, `CommandSequence.objects`, `CowrieSession.objects` calls
   - Added dependency injection via `__init__(ioc_repo=None, cowrie_repo=None)`

2. **FireHolCron** (`greedybear/cronjobs/firehol.py`)
   - Now uses `FireHolRepository`
   - Replaced direct `FireHolList.objects` calls
   - Added dependency injection via `__init__(firehol_repo=None)`

3. **MassScannersCron** (`greedybear/cronjobs/mass_scanners.py`)
   - Now uses `MassScannerRepository` and `IocRepository`
   - Replaced direct `MassScanner.objects` and `IOC.objects` calls
   - Added dependency injection via `__init__(mass_scanner_repo=None, ioc_repo=None)`
   - **Fixed logging to occur after DB transaction completes**

### Testing:
- Added 16 comprehensive new tests covering all new repository methods
- Tests include edge cases (empty results, missing entries, etc.)
- All 359 tests passing (verified in Docker environment)

## Related issues

Addresses #633 (Phase 2)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue) - Fixed logging order in mass_scanners.py
- [x] New feature (non-breaking change which adds functionality) - Repository pattern implementation
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist

- [x] I have read and understood the rules about [how to Contribute](https://intelowlproject.github.io/docs/GreedyBear/Contribute/) to this project.
- [x] The pull request is for the branch `develop`.
- [ ] I have added documentation of the new features.
- [x] Linter (`Ruff`) gave 0 errors. Pre-commit hooks passed.
- [x] I have added tests for the feature/bug I solved. All the tests (new and old ones) gave 0 errors. (359/359 tests passing)
- [ ] If changes were made to an existing model/serializer/view, the docs were updated and regenerated.
- [ ] If the GUI has been modified: N/A (backend only changes)

### CODE REVIEW

**Comment by opbot-xd** [comment]
## Note: Logging Issue in whatsmyip.py (Out of Scope)

During development, I noticed a logging inconsistency in `greedybear/cronjobs/whatsmyip.py` where the log statement occurs **before** the database transaction completes:

```python
# Current code (line 18):
self.log.info(f"added new whatsmyip domain {domain=}")
WhatsMyIPDomain(domain=domain).save()
```

This should be:
```python
WhatsMyIPDomain(domain=domain).save()
self.log.info(f"added new whatsmyip domain {domain=}")

**Comment by mlodic** [comment]
this is a minor change, feel free to add it directly here. Thanks for reporting

**Comment by regulartim** [comment]
Looks good at first glance. :+1:  I will review tomorrow!

**Comment by opbot-xd** [comment]
I noticed that GitHub Copilot automatically requested a review on this PR. Is this a new automated workflow in the repository settings? Just checking to ensure I didn't trigger anything by accident!

**Comment by mlodic** [comment]
yes, I just enabled it as as experimental

File: greedybear/cronjobs/repositories/mass_scanner.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        Args:
+            ip_address: IP address to look up.
+
+        Returns:
+            The matching MassScanner entry, or None if not found.
+        """
+        try:
+            return MassScanner.objects.get(ip_address=ip_address)
+        except MassScanner.DoesNotExist:
+            return None
```
Comment (regulartim) [inline greedybear/cronjobs/repositories/mass_scanner.py]: Sorry, maybe this is stupid but where do we need this method? I can't find any reference (except for tests).

File: greedybear/cronjobs/repositories/mass_scanner.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        """
+        Check if a mass scanner entry exists for the given IP.
+
+        Args:
+            ip_address: IP address to check.
+
+        Returns:
+            True if the entry exists, False otherwise.
+        """
+        return MassScanner.objects.filter(ip_address=ip_address).exists()
```
Comment (regulartim) [inline greedybear/cronjobs/repositories/mass_scanner.py]: Can these three method be condensed to a single `get_or_create` method?

**Review by regulartim — CHANGES_REQUESTED** [review]
Excellent! 👍 I only found some minor, cosmetic stuff.

**Comment by regulartim** [comment]
Do you think it is time to split up the `test_repositories.py` ? It will have almost 1000 lines after this merge. 
(But not in this PR. I would open a separate issue for that.)

File: greedybear/cronjobs/repositories/mass_scanner.py (hunk 1 above)
Comment (opbot-xd) [reply greedybear/cronjobs/repositories/mass_scanner.py]: You're absolutely right! I added `get_by_ip()` following a common repository pattern (providing basic CRUD operations), but the actual `MassScannersCron` only uses `exists()`.

Should I remove `get_by_ip()` from `MassScannerRepository` since it's not needed? I can keep just the methods that are actually used: `exists()`, `create()`, and `save()`.

Let me know your preference!

File: greedybear/cronjobs/repositories/mass_scanner.py (hunk 2 above)
Comment (opbot-xd) [reply greedybear/cronjobs/repositories/mass_scanner.py]: Absolutely! That's a much cleaner approach. 

I can refactor `MassScannerRepository` to use a single `get_or_create()` method (similar to how `FireHolRepository` works), which will:
1. Check if the entry exists
2. Return it if found
3. Create and return it if not found

This will also simplify the calling code in `MassScannersCron` from:
```python
if not self.mass_scanner_repo.exists(ip_address):
    self.mass_scanner_repo.create(ip_address, reason)
```

To just:
```python
self.mass_scanner_repo.get_or_create(ip_address, reason)
```

File: greedybear/cronjobs/repositories/mass_scanner.py (hunk 1 above)
Comment (regulartim) [reply greedybear/cronjobs/repositories/mass_scanner.py]: I get your point, but I would rather not have unused methods in the code base. So I would suggest to remove it.

**Comment by opbot-xd** [comment]
Great idea! I agree that `test_repositories.py` is getting quite large. 

Splitting it up would make it more maintainable. A logical split could be:
- `test_ioc_repository.py` - IOC-related tests
- `test_cowrie_session_repository.py` - Cowrie session tests  
- `test_sensor_repository.py` - Sensor tests
- `test_elastic_repository.py` - Elasticsearch tests
- `test_firehol_repository.py` - FireHol tests
- `test_mass_scanner_repository.py` - Mass scanner tests

I'm happy to tackle that refactoring in a separate PR once this one is merged.

**Comment by opbot-xd** [comment]
Done! I've refactored `MassScannerRepository` to use a single `get_or_create()` method.

**Changes:**
- Removed `get_by_ip()`, `create()`, `save()`, and `exists()`
- Replaced with `get_or_create()` following the same pattern as `FireHolRepository`
- Updated `MassScannersCron` to use the simpler API
- Updated tests accordingly

**Comment by opbot-xd** [comment]
Thanks for the Copilot review!

Regarding the wildcard import warnings for `firehol.py` and `mass_scanner.py` not defining `__all__`:

This is **intentionally ignored** in this codebase. The project's Ruff configuration (`.github/configurations/python_linters/.ruff.toml`) explicitly ignores **F403** (wildcard imports):

```toml
ignore = [
    # F403: Allow wildcard imports in __init__.py files
    "F403",
    ...
]
```

Following the established codebase patterns here.

**Comment by regulartim** [comment]
> This is intentionally ignored in this codebase. The project's Ruff configuration (.github/configurations/python_linters/.ruff.toml) explicitly ignores F403 (wildcard imports):

@mlodic Do you think we can make Copilot respect our Ruff config?
